samples/SampleTypeConversionTest.py

```diff
@@ -0,0 +1,138 @@
+        print("Can't set CUDA DLLs search path.", file = sys.stderr)
+        exit(1)
+
+    # Add PATH as well for minor CUDA releases
+    sys_path = os.environ["PATH"]
+    if sys_path:
+        paths = sys_path.split(';')
+        for path in paths:
+            if os.path.isdir(path):
+                os.add_dll_directory(path)
+    else:
+        print("PATH environment variable is not set.", file = sys.stderr)
+        exit(1)
+
+import pycuda.driver as cuda
+import PyNvCodec as nvc
+import numpy as np
+import torch
+import nvcv
+
+def decode(gpuID, encFilePath, decFilePath):
+    cuda.init()
+    cuda_ctx = cuda.Device(gpuID).retain_primary_context()
+    cuda_ctx.push()
+    cuda_str = cuda.Stream()
+    cuda_ctx.pop()
+
+    decFile = open(decFilePath, "wb")
+
+    nvDmx = nvc.PyFFmpegDemuxer(encFilePath)
+    nvDec = nvc.PyNvDecoder(nvDmx.Width(), nvDmx.Height(), nvDmx.Format(), nvDmx.Codec(), cuda_ctx.handle, cuda_str.handle)
+
+    nvDwn = nvc.PySurfaceDownloader(nvDmx.Width(), nvDmx.Height(), nvDmx.Format(), cuda_ctx.handle, cuda_str.handle)
+
+    res = str(nvDmx.Width()) + 'x' + str(nvDmx.Height())
+
+    nvEnc = nvc.PyNvEncoder({'preset': 'P5', 'tuning_info': 'high_quality', 'codec': 'h264',
+                             'profile': 'high', 's': res, 'bitrate': '10M'}, cuda_ctx.handle, cuda_str.handle)
+
+    packet = np.ndarray(shape=(0), dtype=np.uint8)
+    frameSize = int(nvDmx.Width() * nvDmx.Height() * 3 / 2)
+    rawFrame = np.ndarray(shape=(frameSize), dtype=np.uint8)
+    pdata_in, pdata_out = nvc.PacketData(), nvc.PacketData()
+
+    encFrame = np.ndarray(shape=(0), dtype=np.uint8)
+
+    # Determine colorspace conversion parameters.
+    # Some video streams don't specify these parameters so default values
+    # are most widespread bt601 and mpeg.
+    cspace, crange = nvDmx.ColorSpace(), nvDmx.ColorRange()
+    if nvc.ColorSpace.UNSPEC == cspace:
+        cspace = nvc.ColorSpace.BT_601
+    if nvc.ColorRange.UDEF == crange:
+        crange = nvc.ColorRange.MPEG
+    cc_ctx = nvc.ColorspaceConversionContext(cspace, crange)
+    print('Color space: ', str(cspace))
+    print('Color range: ', str(crange))
+
+    while True:
+        # Demuxer has sync design, it returns packet every time it's called.
+        # If demuxer can't return packet it usually means EOF.
+        torch.cuda.nvtx.range_push("DemuxSinglePacket")
+        success = nvDmx.DemuxSinglePacket(packet)
+        torch.cuda.nvtx.range_pop()
+        if not success:
+            break
+
+
+        # Get last packet data to obtain frame timestamp
+        torch.cuda.nvtx.range_push("LastPacketData")
+        nvDmx.LastPacketData(pdata_in)
+        torch.cuda.nvtx.range_pop()
+
+        # Decoder is async by design.
+        # As it consumes packets from demuxer one at a time it may not return
+        # decoded surface every time the decoding function is called.
+        torch.cuda.nvtx.range_push("nvDec.DecodeSurfaceFromPacket")
```

Comment:
maybe we could have our one internal `__nvtx_range_push()`/`__nvtx_range_pop` that doesn't depend on torch.